test/fn/move-contacts.spec.js

```diff
@@ -0,0 +1,302 @@
+        writtenDocs.length = 0;
+      });
+
+      afterEach(async () => pouchDb.destroy());
+
+      it('move health_center_1 to district_2', async () => {
+        await updateLineagesAndStage({
+          contactIds: ['health_center_1'],
+          parentId: 'district_2',
+        }, pouchDb);
+
+        expect(await getWrittenDoc('health_center_1_contact')).to.deep.eq({
+          _id: 'health_center_1_contact',
+          type: 'person',
+          parent: parentsToLineage('health_center_1', 'district_2'),
+        });
+
+        expect(await getWrittenDoc('health_center_1')).to.deep.eq({
+          _id: 'health_center_1',
+          type: 'health_center',
+          contact: { _id: 'health_center_1_contact' },
+          parent: parentsToLineage('district_2'),
+        });
+
+        expect(await getWrittenDoc('clinic_1')).to.deep.eq({
+          _id: 'clinic_1',
+          type: 'clinic',
+          contact: { _id: 'clinic_1_contact' },
+          parent: parentsToLineage('health_center_1', 'district_2'),
+        });
+
+        expect(await getWrittenDoc('patient_1')).to.deep.eq({
+          _id: 'patient_1',
+          type: 'person',
+          parent: parentsToLineage('clinic_1', 'health_center_1', 'district_2'),
+        });
+
+        expect(await getWrittenDoc('report_1')).to.deep.eq({
+          _id: 'report_1',
+          form: 'foo',
+          type: 'data_record',
+          contact: parentsToLineage('health_center_1_contact', 'health_center_1', 'district_2'),
+        });
+      });
+
+      it('move health_center_1 to root', async () => {
+        await updateLineagesAndStage({
+          contactIds: ['health_center_1'],
+          parentId: 'root',
+        }, pouchDb);
+
+        expect(await getWrittenDoc('health_center_1_contact')).to.deep.eq({
+          _id: 'health_center_1_contact',
+          type: 'person',
+          parent: parentsToLineage('health_center_1'),
+        });
+
+        expect(await getWrittenDoc('health_center_1')).to.deep.eq({
+          _id: 'health_center_1',
+          type: 'health_center',
+          contact: { _id: 'health_center_1_contact' },
+          parent: parentsToLineage(),
+        });
+
+        expect(await getWrittenDoc('clinic_1')).to.deep.eq({
+          _id: 'clinic_1',
+          type: 'clinic',
+          contact: { _id: 'clinic_1_contact' },
+          parent: parentsToLineage('health_center_1'),
+        });
+
+        expect(await getWrittenDoc('patient_1')).to.deep.eq({
+          _id: 'patient_1',
+          type: 'person',
+          parent: parentsToLineage('clinic_1', 'health_center_1'),
+        });
+
+        expect(await getWrittenDoc('report_1')).to.deep.eq({
+          _id: 'report_1',
+          form: 'foo',
+          type: 'data_record',
+          contact: parentsToLineage('health_center_1_contact', 'health_center_1'),
+        });
+      });
+
+      it('move district_1 from root', async () => {
```

Comment:
That wasn't really my intent here, since I was not previously enforcing any hierarchy in this case. The test is intended to test the case where a document has no parent and it gets move so it now has a parent.